sakiparens.py

```diff
@@ -0,0 +1,17 @@
+from __future__ import unicode_literals
+
+def parens(string):
+    stack = []
+    for char in string:
+        if char == '(':
+            stack.append(char)
+        elif char == ')':
+            try:
+                stack.pop()
+            except:
```

Comment:
Sure I meant IndexError. I'will fix it.

On Sun, Jun 28, 2015 at 4:22 PM, Cris Ewing <notifications@github.com>
wrote:

> In sakiparens.py
> <https://github.com/ndraper2/data-structures/pull/4#discussion_r33428328>:
>
> > @@ -0,0 +1,17 @@
> > +from __future__ import unicode_literals
> > +
> > +def parens(string):
> > +    stack = []
> > +    for char in string:
> > +        if char == '(':
> > +            stack.append(char)
> > +        elif char == ')':
> > +            try:
> > +                stack.pop()
> > +            except:
>
> You probably want to catch a specific exception. What is raised by the
> stack if .pop() is called on an empty stack? That's the error you should
> be catching, not just any error.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ndraper2/data-structures/pull/4/files#r33428328>.
>